src/sif/autocost.cc

```diff
@@ -295,6 +317,10 @@ class AutoCost : public DynamicCost {
   virtual bool Allowed(const baldr::DirectedEdge* edge,
                        const graph_tile_ptr& tile,
                        uint16_t disallow_mask = kDisallowNone) const override {
+
+    if (!EvaluateHOVAllowed(edge))
```

Comment:
@ktatterso shouldn't we be calling `EvaluateHOVAllowed` in `AutoCost::Allowed` and `AutoCost::AllowedReverse` versus here as the comment says, "Function to be used in location searching..."  Bidirectional_astar calls it from [here](https://github.com/valhalla/valhalla/blob/master/src/thor/bidirectional_astar.cc#L260-L261) and [here](https://github.com/valhalla/valhalla/blob/master/src/thor/bidirectional_astar.cc#L267-L268)

Reply:
@gknisely Good catch.  My testing was limited so I missed this.  I've fixed the `Allowed()` and `AllowedReverse()` methods that are called by BidirAstar to check for HOV routability.  I've also added some new tests that capture this fix.

Reply:
@ktatterso can you move !EvaluateHOVAllowed(edge) to the huge if statement.  just to be consistent

Reply:
@gknisely I can but... I hate those big conditionals... they're hard to mentally dissect and also hard to debug.  :/

Reply:
@gknisely ok I pushed the changes you requested.  Still waiting on the data to build...